sdcm/cluster.py

```diff
@@ -3661,25 +3661,35 @@ def node_run_stress_bench(node, loader_idx, stress_cmd, node_list):
                                          (loader_idx, uuid.uuid4()))
             ips = ",".join([n.private_ip_address for n in node_list])
             bench_log = tempfile.NamedTemporaryFile(prefix='scylla-bench-', suffix='.log').name
-            result = node.remoter.run(cmd="/$HOME/go/bin/{} -nodes {} | tee {}".format(stress_cmd.strip(), ips, bench_log),
+
+            result = node.remoter.run(cmd="/$HOME/go/bin/{name} -nodes {ips} |tee {log}".format(name=stress_cmd.strip(),
+                                                                                                ips=ips,
+                                                                                                log=bench_log),
                                       timeout=timeout,
                                       ignore_status=True,
                                       log_file=log_file_name)
 
-            CassandraStressEvent(type='finish', node=str(node), stress_cmd=stress_cmd, log_file_name=log_file_name)
+            ScyllaBenchEvent(type='finish', node=str(node), stress_cmd=stress_cmd, log_file_name=log_file_name)
 
             _queue[RES_QUEUE].put((node, result))
             _queue[TASK_QUEUE].task_done()
 
-        loaders_to_use = self.nodes if not use_single_loader else self.nodes[:1]  # pylint: disable=no-member
-        for loader_idx, loader in enumerate(loaders_to_use):
+        def add_and_start_thread(args):
             setup_thread = threading.Thread(target=node_run_stress_bench,
-                                            args=(loader, loader_idx,
-                                                  stress_cmd, node_list))
+                                            args=args)
             setup_thread.daemon = True
             setup_thread.start()
             time.sleep(30)
 
+        if round_robin:
+            loader = self.get_loader()
+            loader_idx = [i for i, loader_obj in enumerate(self.nodes) if loader_obj == loader][0]
+            add_and_start_thread(args=(loader, loader_idx, stress_cmd, node_list))
```

Comment:
Does this code works when you have more commands than loader?
less commands than loaders?
Not that readable to me.
Basically you could just have list of commands, num of loaders, each loader gets the MOD command that fits its index.

Reply:
Longevity test send stress cmds one by one. To do it on the way that you suggest - it means to change longevity mechanism. We can do it as separate task.
Another solution - realize it on similat way as CassandraStressThread /GeminiThread

Reply:
ok.